src/screens/SquareScreen.js

```diff
@@ -0,0 +1,84 @@
+import React, { useReducer } from "react";
+import { StyleSheet, View } from "react-native";
+import ColorCounter from "../components/ColorCounter";
+
+const COLOR_INCREMENT = 15;
+const initialState = { red: 0, green: 0, blue: 0 };
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "change_red":
+      return state.red + action.payload > 255 || state.red + action.payload < 0
+        ? state
+        : { ...state, red: state.red + action.payload };
+    case "change_green":
+      return state.green + action.payload > 255 ||
+        state.green + action.payload < 0
+        ? state
+        : { ...state, green: state.green + action.payload };
+    case "change_blue":
+      return state.blue + action.payload > 255 ||
+        state.blue + action.payload < 0
+        ? state
+        : { ...state, blue: state.blue + action.payload };
+    default:
+      return state;
+  }
+};
+
+const SquareScreen = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const getColor = () => {
+    const { red, green, blue } = state;
+    return {
+      ...styles.colorBox,
+      backgroundColor: `rgb(${red}, ${green}, ${blue})`,
+    };
+  };
+
+  return (
+    <View>
+      <ColorCounter
+        onIncrease={() =>
+          dispatch({ type: "change_red", payload: COLOR_INCREMENT })
+        }
+        onDecrease={() =>
+          dispatch({ type: "change_red", payload: -COLOR_INCREMENT })
+        }
+        color="Red"
+      />
+      <ColorCounter
+        onIncrease={() =>
+          dispatch({ type: "change_green", payload: COLOR_INCREMENT })
+        }
+        onDecrease={() =>
+          dispatch({ type: "change_green", payload: -COLOR_INCREMENT })
+        }
+        color="Green"
+      />
+      <ColorCounter
+        onIncrease={() =>
+          dispatch({ type: "change_blue", payload: COLOR_INCREMENT })
+        }
+        onDecrease={() =>
+          dispatch({ type: "change_blue", payload: -COLOR_INCREMENT })
+        }
+        color="Blue"
+      />
+
+      <View style={getColor()}></View>
```

Comment:
you may not need an extra method. do something like this:

```typescript
const bgColor = `rgb(${red}, ${green}, ${blue})`
<View style={...styles.colorBox,  backgroundColor: bgColor} />
```

or
```typescript
<View style={...styles.colorBox,  backgroundColor: `rgb(${red}, ${green}, ${blue})`} />
```

The idea is to have two things:

1. when I look at the jsx section i should only see jsx and when I look at the styles I should only see css.
2. when I want to change a style of something I shouldn't be jumping back and forth between styles and in-line styles.
3. Yes, backGround color is dynamic. That can stay along with jsx, because you have no other solution.